internal/datasource/host/info.go

```diff
@@ -0,0 +1,348 @@
+}
+
+func getContainerIDFromPatterns(word string) string {
+	slices := scopePattern.FindStringSubmatch(word)
+	if containsContainerID(slices) {
+		return slices[1]
+	}
+
+	slices = basePattern.FindStringSubmatch(word)
+	if containsContainerID(slices) {
+		return slices[1]
+	}
+
+	slices = colonPattern.FindStringSubmatch(word)
+	if containsContainerID(slices) {
+		return slices[1]
+	}
+
+	slices = containersPattern.FindStringSubmatch(word)
+	if containsContainerID(slices) {
+		return slices[1]
+	}
+
+	slices = containerdPattern.FindStringSubmatch(word)
+	if containsContainerID(slices) {
+		return slices[1]
+	}
+
+	return ""
+}
+
+func containsContainerID(slices []string) bool {
+	return len(slices) >= 2 && len(slices[1]) == lengthOfContainerID
+}
+
+func (i *Info) getHostID(ctx context.Context) string {
+	res, err, _ := singleflightGroup.Do(GetSystemUUIDKey, func() (interface{}, error) {
+		var err error
+
+		hostID, err := i.exec.GetHostID(ctx)
+		if err != nil {
+			slog.WarnContext(ctx, "Unable to get host ID", "error", err)
+			return "", err
+		}
+
+		return uuid.NewMD5(uuid.Nil, []byte(hostID)).String(), err
+	})
+
+	if err != nil {
+		slog.WarnContext(ctx, "Unable to get host ID", "error", err)
+		return ""
+	}
+
+	if result, ok := res.(string); ok {
+		return result
+	}
+
+	return ""
+}
+
+func (i *Info) getReleaseInfo(ctx context.Context, osReleaseLocation string) (releaseInfo *v1.ReleaseInfo) {
+	hostReleaseInfo := i.exec.GetReleaseInfo(ctx)
+	osRelease, err := readOsRelease(osReleaseLocation)
+	if err != nil {
+		slog.WarnContext(ctx, "Unable to read from os release file", "error", err)
+
+		return hostReleaseInfo
+	}
+
+	return mergeHostAndOsReleaseInfo(hostReleaseInfo, osRelease)
+}
+
+func readOsRelease(path string) (map[string]string, error) {
+	f, err := os.Open(path)
+	if err != nil {
+		return nil, fmt.Errorf("release file %s is unreadable: %w", path, err)
+	}
+	defer func(f *os.File, fileName string) {
+		closeErr := f.Close()
+		if closeErr != nil {
+			slog.Error("Unable to close file", "file", fileName, "error", closeErr)
+		}
+	}(f, path)
+
+	info, err := parseOsReleaseFile(f)
+	if err != nil {
+		return nil, fmt.Errorf("release file %s is unparsable: %w", path, err)
+	}
+
+	return info, nil
+}
+
+func parseOsReleaseFile(reader io.Reader) (map[string]string, error) {
+	osReleaseInfoMap := map[string]string{"NAME": "unix"}
+	scanner := bufio.NewScanner(reader)
+	for scanner.Scan() {
+		line := strings.TrimSpace(scanner.Text())
+		field := strings.Split(line, "=")
+		if len(field) < 2 {
+			continue
+		}
+		osReleaseInfoMap[field[0]] = strings.Trim(field[1], "\"")
+	}
+
+	if err := scanner.Err(); err != nil {
+		return nil, fmt.Errorf("could not parse os release file %w", err)
+	}
+
+	return osReleaseInfoMap, nil
+}
+
+func mergeHostAndOsReleaseInfo(releaseInfo *v1.ReleaseInfo,
+	osReleaseInfo map[string]string,
+) (release *v1.ReleaseInfo) {
+	if len(osReleaseInfo[versionID]) == 0 {
```

Comment:
consider using a switch statement for this?

Reply:
Not sure how this could be put in a `switch` 🤔

Reply:
cant use switch statement here